File: 26_remove_duplicates_from_sorted_array/yang.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        index = 0
        for num in nums:
            if num > nums[index]:
                index += 1
                nums[index] = num

        return index

File: 0026_Remove_Duplicates_from_Sorted_Array/yang.cpp
#include <iostream>
#include <vector>


int removeDuplicates(std::vector<int>& nums)
{
    int index = 0;
    if (nums.size() == 0)
    {
        return 0;
    }
    for (auto & element: nums)
    {
        if (element > nums[index])
        {
            index += 1;
            nums[index] = element;
        }

    }
    return index+1;
}


int main()
{
    std::vector<int> v = {1, 1, 2, 2, 3, 4};
    int duplicates = removeDuplicates(v);
    std::cout << duplicates << std::endl;

    return 0;
}
